pkg/options/options.go

```diff
@@ -0,0 +1,137 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package options
+
+import (
+	"fmt"
+
+	"github.com/spf13/pflag"
+	"github.com/xmudrii/etcdproxy-controller/pkg/cmd/controller/config"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// CoreEtcdOptions type is used to wire the core etcd information used by controller to create ReplicaSets.
+type CoreEtcdOptions struct {
+	URL             string
+	CAConfigMapName string
+	CertSecretName  string
+}
+
+// EtcdProxyOptions type is used to pass information from cli to the controller.
+type EtcdProxyOptions struct {
+	// CoreEtcdOptions contains information needed to wire up ReplicaSets and the core etcd.
+	CoreEtcdOptions *CoreEtcdOptions
+
+	// ControllerNamespace is name of namespace where controller is deployed.
+	ControllerNamespace string
+
+	// KubeconfigPath is used to obtain path to kubeconfig, used to create kubeclients.
+	KubeconfigPath string
+
+	// ProxyImage is name of the etcd image to be used for etcd-proxy ReplicaSet creation.
+	ProxyImage string
+}
+
+func NewCoreEtcdOptions() *CoreEtcdOptions {
+	return &CoreEtcdOptions{}
+}
+
+func NewEtcdProxyOptions() *EtcdProxyOptions {
+	return &EtcdProxyOptions{
+		CoreEtcdOptions: NewCoreEtcdOptions(),
+	}
+}
+
+func (e *EtcdProxyOptions) AddFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&e.CoreEtcdOptions.URL, "etcd-core-url", "", "The address of the core etcd server.")
+	fs.StringVar(&e.CoreEtcdOptions.CAConfigMapName, "etcd-core-ca-configmap", "etcd-coreserving-ca", "The name of the ConfigMap where CA is stored.")
```

Comment:
if you set defaults in the actual object, then this becomes `fs.StringVar(&e.CoreEtcdOptions.CAConfigMapName, "etcd-core-ca-configmap", e.CoreEtcdOptions.CAConfigMapName, "The name of the ConfigMap where CA is stored.")` and someone can override them.

Reply:
That sounds much better. Updated.